freud/diffraction.pyx

```diff
@@ -0,0 +1,348 @@
+        Returns
+        -------
+        numpy.ndarray (N,N), scaled array
+        """
+        ny, nx = np.shape(a)
+        y = np.array([list(range(ny))])
+        x = np.array([list(range(nx))])
+        d = interpolate.RectBivariateSpline(x, y, a, kx=1, ky=1)
+        x = np.linspace(0, nx, self.N)
+        y = np.linspace(0, ny, self.N)
+        d = d(x, y)
+        return d
+
+    def shear_back(self, img, inv_shear):
+        """
+        TODO
+        Parameters
+        ----------
+        img : numpy.ndarray (N,N), array of diffraction intensities
+        inv_shear : numpy.ndarray (2,2), inverse shear matrix
+        Returns
+        -------
+        numpy.ndarray (N,N), sheared array of diffraction intensities
+        """
+        roll = img.shape[0] / 2 - 1
+        ss = np.max(self.box) * inv_shear
+        A1 = np.array([[1, 0, -roll],
+                      [0, 1, -roll],
+                      [0, 0, 1]])
+
+        A2 = np.array([[ss[1, 0], ss[0, 0], roll],
+                      [ss[1, 1], ss[0, 1], roll],
+                      [0, 0, 1]])
+
+        A3 = np.linalg.inv(np.dot(A2, A1))
+        A4 = A3[0:2, 0:2]
+        A5 = A3[0:2, 2]
+        img = ndimage.interpolation.affine_transform(img, A4, A5,
+                                                     mode="constant")
+        return img
+
+    def compute(self, system, rot, cutout=True):
+        """
+        2D FFT to get diffraction pattern from intensity matrix.
+        Parameters
+        ----------
+        rot: numpy.ndarray (3, 3), rotation matrix
+        cutout: bool, return diffraction pattern with circle cutout
+        (default True)
+        Returns
+        -------
+        numpy.ndarray (N,N), diffraction pattern
+        """
+        box, positions = system
+        box = box.to_matrix()
+        N = self.N / self.zoom
+        inv_shear = self.calc_proj(rot)
+        xy = np.copy(rowan.rotate(rot, self.orig)[:, 0:2])
+        xy = np.dot(xy, inv_shear.T)
+        xy = self.pbc_2d(xy, N)
+        im = self.bin(xy, N)
+
+        self.dp = np.fft.fft2(im)
+        self.dp = ndimage.fourier.fourier_gaussian(self.dp,
+                                                   self.peak_width / self.zoom)
+        self.dp = np.fft.fftshift(self.dp)
+        self.dp = np.absolute(self.dp)
+        self.dp *= self.dp
+
+        self.dp = self.scale(self.dp)
+        self.dp = self.shear_back(self.dp, inv_shear)
+        self.dp /= self.dp.max()
+        self.dp[self.dp < self.bot] = self.bot
+        self.dp[self.dp > self.top] = self.top
+        self.dp = np.log10(self.dp)
+
+        """
+        NOTE: cut into a circle, not sure if needed-YJ
+        """
+        # if not cutout:
+        #     return dp
+
+        # idbig = self.circle_cutout(dp)
+        # dp[np.unravel_index(idbig, (self.N, self.N))] = np.log(self.bot)
+        return self.dp
+
+    def diffraction_pattern(self):
+        return self.dp
+
+    def plot(self, ax=None):
```

Comment:
You'll want to define the plotting logic in `plot.py` and then import and call that method in `Diffraction.plot`. We don't require matplotlib as a hard dependency, and this structure allows us to use matplotlib as a "soft" dependency and raise errors if it's not available.

Also add methods for `__repr__` and `_repr_png_` like other compute objects in freud.